sui_core/src/safe_client.rs

```diff
@@ -190,19 +189,33 @@ impl<C> SafeClient<C> {
             .check(&signed_batch.batch, signed_batch.authority)?;
 
         // ensure transactions enclosed match requested range
-        fp_ensure!(
-            signed_batch.batch.initial_sequence_number >= request.start
-                && signed_batch.batch.next_sequence_number
-                    <= (request.end + signed_batch.batch.size),
-            SuiError::ByzantineAuthoritySuspicion {
-                authority: self.address
-            }
-        );
+        /*
+
+        TODO: check that the batch is within bounds given that the
+              bounds may now not be known by the requester.
+
+        if let Some(start) = &request.start {
+            fp_ensure!(
+                signed_batch.batch.initial_sequence_number >= *start
+                    && signed_batch.batch.next_sequence_number
+                        <= (*start + request.length + signed_batch.batch.size),
+                SuiError::ByzantineAuthoritySuspicion {
+                    authority: self.address
+                }
+            );
+        }
+        */
```

Comment:
Mind using line comments `//` for comment things out over block comments? Block comments make it difficult to know if code is live or dead when grepping

Reply:
converted